### PR TITLE
PR from abrombo/galgebra:master for reference (with merge conflicts solved)

### PR DESCRIPTION
**Pull request by eric-wieser**
A variant of #52 containing the ~~(trivial)~~ manual merge(s) needed to make the diff readable

### CODE REVIEW

**Comment by eric-wieser** [comment]
Created https://github.com/abrombo/galgebra/pull/2 to get this merge into @abrombo's hands

File: galgebra/mv.py [hunk 1]
```diff
@@ -2473,6 +2493,11 @@ def Fmt(self, fmt=1, title=None, dop_fmt=None):
                 return s
         return
 
+    def odot(self,dot_flg=True):
+        new_self = copy.deepcopy(self)
+        new_self.dot_flg = dot_flg
+        return new_self
```
Comment (eric-wieser) [inline galgebra/mv.py]: This is the only change to the differential operators part of this file (xref gh-133)

**Comment by eric-wieser** [comment]
Rebased with some flake8 fixes and removals of stray files.

I think all that's really in here now is an incomplete `odot` implementation, and the printer stuff.

File: galgebra/mv.py [hunk 2]
```diff
@@ -1306,13 +1306,22 @@ def simplify(self, modes=simplify):
...
+
+    """
     def subs(self, d):
         # For each scalar coef of the multivector apply substitution argument d
         obj = sum((
             coef.subs(d) * base for coef, base in metric.linear_expand_terms(self.obj)
         ), S(0))
         return Mv(obj, ga=self.Ga)
-
+    """
```
Comment (eric-wieser) [inline galgebra/mv.py:1324]: Covered by https://github.com/pygae/galgebra/pull/243 I think.

This implementation is wrong, and mutates `self.obj`, something which is a bad idea.

File: galgebra/printer.py [hunk 3]
```diff
@@ -983,20 +1100,27 @@ def Format(Fmode=True, Dmode=True, dop=1, inverse='full'):
...
         Basic.__repr__ = lambda self: GaLatexPrinter().doprint(self)
         Matrix.__repr__ = lambda self: GaLatexPrinter().doprint(self)
 
-        if isinteractive():
+        if isinteractive():  # Set up for Jupyter Notebook/Lab
             init_printing(use_latex= 'mathjax')
+            from IPython.core.interactiveshell import InteractiveShell
+            #  Allow multiple outputs in an output cell
+            #  https://forums.fast.ai/t/jupyter-notebook-enhancements-tips-and-tricks/17064/2
+            InteractiveShell.ast_node_interactivity = "all"
```
Comment (eric-wieser) [inline galgebra/printer.py:1121]: This isn't about allowing multiple outputs, it's about producing them automatically.
The user can always print multiple outputs with `display(...)` if they want to.

This doesn't seem like a good idea to me. If users want it, they can do it themselves.

File: galgebra/printer.py [hunk 4]
```diff
@@ -963,6 +1031,55 @@ def latex(expr, **settings):
...
+
+    s = list(s)
+
+    GaLatexPrinter.fmt_dict = {'t': False, 'h': False}
+
+    if isinstance(s[0], str):
+
+        if s[0] == 'h':
+            GaLatexPrinter.fmt_dict['h'] = True
+            s = s[1:]
```
Comment (eric-wieser) [inline galgebra/printer.py:1044]: This looks like a bad idea to me. This changes the behavior of `print(my_str)` to print nothing at all if `my_str == 'h'`!

**Comment by eric-wieser** [comment]
Conflicts solved again

File: galgebra/printer.py [hunk 5]
```diff
@@ -47,15 +49,77 @@
 \newcommand{\pdiff}[2]{\bfrac{\partial {#1}}{\partial {#2}}}
 \newcommand{\npdiff}[3]{\bfrac{\partial^{#3} {#1}}{\partial {#2}^{#3}}}
 \newcommand{\lbrc}{\left \{}
-\newcommand{\rbrc}{\right \}}
+\newcommand{\rbrc}{\right \]}
+\newcommand{\lbrk}{\left \[}
+\newcommand{\rbrk}{\right \}}
```
Comment (eric-wieser) [inline galgebra/printer.py:54]: @brombo, this looks to me like a bug?

Comment (utensil) [reply galgebra/printer.py:54]: Yeah, obviously

**Comment by utensil** [comment]
Thank you @abrombo @eric-wieser , this should be superseded by #510 which merged gprinter from upstream. Feel free to reopen if there are remaining issues.

**Pull request closed without merging**